awx/main/scheduler/task_manager.py

```diff
@@ -481,6 +450,145 @@ def generate_dependencies(self, undeped_tasks):
 
         return created_dependencies
 
+    def process_tasks(self, all_sorted_tasks):
+        self.generate_dependencies(all_sorted_tasks)
+        self.subsystem_metrics.inc(f"{self.prefix}_pending_processed", len(all_sorted_tasks))
+
+    @timeit
+    def _schedule(self):
+        all_sorted_tasks = self.get_tasks(dict(status__in=["pending"], dependencies_processed=False))
+
+        if len(all_sorted_tasks) > 0:
+            self.all_inventory_sources = self.get_inventory_source_tasks(all_sorted_tasks)
```

Comment:
For performance, but also for code sanity, I think it would make most sense to create coherent in-memory data structures to lookup inventory sources... and repeat that pattern for projects as well (kind of a custom prefetch_related). This particular assignment is a big throw-my-computer-out-the-window kind of moment for me.